app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb

```diff
@@ -0,0 +1,146 @@
+
+        def jwt_token(authenticator_input)
+          @extract_token_from_credentials.call(
+            credentials: authenticator_input.request.body.read
+          )
+        end
+
+        def jwt_identity_from_request
+          @jwt_identity_from_request ||= identity_provider.jwt_identity
+        end
+
+        def identity_provider
+          @identity_provider ||= create_identity_provider.call(
+            authentication_parameters: @authentication_parameters
+          )
+        end
+
+        def validate_and_decode_token_instance
+          return @validate_and_decode_token_instance if @validate_and_decode_token_instance
+
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreateValidateAndDecodeTokenInstance.new)
+          @validate_and_decode_token_instance = @validate_and_decode_token_class.new(
+            fetch_jwt_claims_to_validate: fetch_jwt_claims_to_validate
+          )
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatedValidateAndDecodeTokenInstance.new)
+          @validate_and_decode_token_instance
+        end
+
+        def fetch_signing_key
+          @fetch_signing_key ||= ::Util::ConcurrencyLimitedCache.new(
+            ::Util::RateLimitedCache.new(
+              fetch_signing_key_interface,
+              refreshes_per_interval: CACHE_REFRESHES_PER_INTERVAL,
+              rate_limit_interval: CACHE_RATE_LIMIT_INTERVAL,
+              logger: @logger
+            ),
+            max_concurrent_requests: CACHE_MAX_CONCURRENT_REQUESTS,
+            logger: @logger
+          )
+        end
+
+        def create_identity_provider
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreateJwtIdentityProviderInstance.new)
+          @create_identity_provider ||= @create_identity_provider
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatedJwtIdentityProviderInstance.new)
+          @create_identity_provider
+        end
+
+        def fetch_signing_key_interface
+          @fetch_signing_key_interface ||= create_signing_key_interface.call(
+            authentication_parameters: @authentication_parameters
+          )
+        end
+
+        def create_signing_key_interface
+          @create_signing_key_interface ||= Authentication::AuthnJwt::SigningKey::CreateSigningKeyFactory.new
+        end
+
+        def fetch_jwt_claims_to_validate
+          @fetch_jwt_claims_to_validate ||= ::Authentication::AuthnJwt::ValidateAndDecode::FetchJwtClaimsToValidate.new
+        end
+
+        def restrictions_from_annotations
+          @restrictions_from_annotations ||= Authentication::ResourceRestrictions::GetServiceSpecificRestrictionFromAnnotation.new
+        end
+
+        def extract_resource_restrictions
+          @extract_resource_restrictions ||= Authentication::ResourceRestrictions::ExtractResourceRestrictions.new(
+            get_restriction_from_annotation: restrictions_from_annotations,
+            ignore_empty_annotations: false
+          )
```

Comment:
this will simplify this class a lot (if we perform this change in the command classes referenced in this class) as you'll be able to set all these command classes as dependencies of this class and just call their `call` method.

Reply:
@jonahx - i'll just raise a point that this class is different than what we have regularly in our code as it's not a command class and not a value object. @tzheleznyak 's team and i agreed to address this after this PR is merged to master because of time issues. if you have a take on this issue feel free to share.

Reply:
lets talk about it